lang/cs/Org.Apache.REEF.IMRU/Examples/MapperCount/MapperCount.cs

```diff
@@ -39,29 +40,49 @@ private MapperCount(IIMRUClient<int, int, int> imruClient)
-                            .Set(IMRUConfiguration<int, int, int>.ReduceFunction,
-                                GenericType<IntSumReduceFunction>.Class)
-                            .Set(IMRUConfiguration<int, int, int>.UpdateFunction,
-                                GenericType<MapperCountUpdateFunction>.Class)
-                            .Set(IMRUConfiguration<int, int, int>.MapInputCodec,
-                                GenericType<IntStreamingCodec>.Class)
-                            .Set(IMRUConfiguration<int, int, int>.MapOutputCodec,
-                                GenericType<IntStreamingCodec>.Class)
-                            .Set(IMRUConfiguration<int, int, int>.ResultCodec, GenericType<IntStreamingCodec>.Class)
-                            .Build())
-                    .SetJobName("MapperCount")
-                    .Build()
-                );
-            return results.First();
+            var jobDefinitionBuilder = new IMRUJobDefinitionBuilder();
+
+            jobDefinitionBuilder.MapFunctionConfiguration = IMRUMapConfiguration<int, int>.ConfigurationModule
+                .Set(IMRUMapConfiguration<int, int>.MapFunction, GenericType<IdentityMapFunction>.Class)
+                .Build();
+
+            jobDefinitionBuilder.UpdateFunctionConfiguration =
+                IMRUUpdateConfiguration<int, int, int>.ConfigurationModule
+                    .Set(IMRUUpdateConfiguration<int, int, int>.UpdateFunction,
+                        GenericType<MapperCountUpdateFunction>.Class)
+                    .Build();
+
+            jobDefinitionBuilder.MapInputCodecConfiguration = IMRUGenericTypeCodecConfiguration<int>.ConfigurationModule
+                .Set(IMRUGenericTypeCodecConfiguration<int>.Codec, GenericType<IntStreamingCodec>.Class)
+                .Build();
+
+            jobDefinitionBuilder.UpdateFunctionCodecsConfiguration = IMRUGenericTypeCodecConfiguration<int>.ConfigurationModule
+                .Set(IMRUGenericTypeCodecConfiguration<int>.Codec, GenericType<IntStreamingCodec>.Class)
+                .Build();
+
+            jobDefinitionBuilder.ReduceFunctionConfiguration = IMRUReduceFunctionConfiguration<int>.ConfigurationModule
+                .Set(IMRUReduceFunctionConfiguration<int>.ReduceFunction, GenericType<IntSumReduceFunction>.Class)
+                .Build();
+
+            jobDefinitionBuilder.PartitionedDatasetConfiguration =
+                RandomDataConfiguration.ConfigurationModule.Set(RandomDataConfiguration.NumberOfPartitions,
+                    numberofMappers.ToString()).Build();
+
+            jobDefinitionBuilder.NumberOfMappers = numberofMappers;
+
+            var results = _imruClient.Submit(jobDefinitionBuilder.SetJobName("MapperCount").Build());
+
+            if (results != null)
+            {
+                return results.First();
+            }
+
+            return -1;
```

Comment:
no thats not true...For REEF IMRU, results will be written directly on to hdfs or local files and will not be sent to the driver or client. Hence, I am assuming that for those cases it would be null.